### DISCUSSION TITLE
[OpenAPI ] How to implement multipart/form-data APIPlayground for file uploads ?

### DISCUSSION THREAD

**Discussion by shivamstaq** [fuma-nama/fumadocs#654] [Q&A] [answered]
Hey everyone, 
I am new to fumadocs.
I have been trying to convert my OpenAPI 3.1 spec to mdx using `fumadocs-openapi`, but the following request data renders a UI which takes in file as a string. 

```
/ingress:
    post:
      summary: Upload a document
      description: Endpoint to upload a document file
      requestBody:
        required: true
        content:
          multipart/form-data:
            schema:
              type: object
              properties:
                file:
                  type: string
                  format: binary
                  description: The document file to upload
```

Can someone help me understand if fumadocs supports the implementation I am looking for ?

**Comment by fuma-nama** [comment, marked as answer]
Not yet, it's still relatively simple as compared to other alternatives like Scalar. I'll be adding more features soon, welcome to submit a feature request for this.

**Reply by fuma-nama** [reply]
Released a small implementation for `multipart/form-data`, please submit a bug report if you've encountered any issues ;)

**Reply by shivamstaq** [reply]
Thanks for the quick response and release. 
The only thing to add would be to have a way to show if a request is multipart form (and also reflect the same in the curl and js code generated for it)

**Reply by fuma-nama** [reply]
kinda difficult for curl, I've implemented it for JavaScript with FormData API

**Comment by devnjw** [comment]
@fuma-nama 
Thank you for `multipart/form-data` release. 🚀

I have one question.
I submitted file on museum demo, but in the developer tools, it looks like the file hasn't been uploaded.

I was expecting the file to be binary in the payload.

Am I thinking of something wrong?

![image](https://github.com/user-attachments/assets/6c7045ec-5fb2-4049-8ff8-ea7ce4bc874d)
